src/assets/js/protocol/protocol-details.js

```diff
@@ -0,0 +1,229 @@
+    }());
+
+    /**
+     * open
+     * @param {String} id - id of the container to open
+     */
+    Details.open = function(id) {
+        var control = doc.querySelector('[aria-controls=' + id + ']');
+        var details = doc.getElementById(id);
+        control.setAttribute('aria-expanded', true);
+        details.setAttribute('aria-hidden', false);
+        details.classList.remove('is-closed');
+    };
+
+    /**
+     * close
+     * @param {String} id - id of the container to close
+     */
+    Details.close = function(id) {
+        var control = doc.querySelector('[aria-controls=' + id + ']');
+        var details = doc.getElementById(id);
+        control.setAttribute('aria-expanded', false);
+        details.setAttribute('aria-hidden', true);
+        details.classList.add('is-closed');
+    };
+
+    /**
+     * toggle
+     * @param {String} id - id of the container to toggle
+     */
+    Details.toggle = function(id) {
+        var details = doc.getElementById(id);
+        var isClosed = details.getAttribute('aria-hidden');
+
+        if(isClosed == 'true') {
+            Details.open(id);
+        } else {
+            Details.close(id);
+        }
+    };
+
+    /**
+     * handleControlActivation
+     * @param {Event} e - event to handle
+     */
+    Details.handleControlActivation = function(e) {
+        var control = e.target;
+        var id = control.getAttribute('aria-controls');
+        Details.toggle(id);
+    };
+
+    /**
+     * initItem
+     * @param {Object} el - Element to place the control inside of
+     * @param {String} selector - Selector for all control wrappers
+     * - assumes every sibling until the next control is associated with the control
+     */
+    Details.initItem = function(el, selector) {
+        var summary = el;
+        var control = doc.createElement('button');
+        var details;
+        var parent = summary.parentNode;
+        var summarySiblings;
+
+        // if it's already been initalized, don't do it again
+        if(summary.querySelectorAll('button').length !== 0) {
+            return;
+        }
+
+        // add class to parent to indicate js initialized
+        parent.classList.add('is-details');
+
+        // Expand
+        // siblings of the summary, until next summary
+        summarySiblings = window.Mozilla.Utils.nextUntil(summary, selector);
+
+        // look to see if all children are already in a wrapper we can use
+        if(summarySiblings.length == 1) {
+            details = summarySiblings[0];
+        } else if (summarySiblings.length > 1){
+            details = doc.createElement('div');
+            details.classList.add('mzp-c-details-wrapper');
+            summarySiblings.forEach(function(sibling) {
+                details.append(sibling);
+            });
+
+            summary.parentNode.insertBefore(details, summary.nextSibling);
+        } else {
+            // error
+            // TODO: what do we do with errors? fail silently?
```

Comment:
What do we do with errors? Fail silently?

Reply:
I guess It depends how bad an error could be. Throwing an error is useful for developers, but it could also halt execution of JS which could be even worse than the original error. Maybe the best thing to do is to try and fail gracefully here?

Reply:
I guess It depends how bad an error could be. Throwing an error is useful for developers, but it could also halt execution of JS which could be even worse than the original error. Maybe the best thing to do is to try and fail gracefully here?

Reply:
I moved stuff around so I could halt initialization if something goes wrong.